source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -177,11 +299,77 @@ def culling_journeys(resp, request):
 
     no remove done in debug
     """
+    logger = logging.getLogger(__name__)
+
     if request['debug']:
         return
 
-    if request["max_nb_journeys"] and len(resp.journeys) > request["max_nb_journeys"]:
-        del resp.journeys[request["max_nb_journeys"]:]
+    if not request["max_nb_journeys"] or request["max_nb_journeys"] > len(resp.journeys):
+        logger.debug('No need to cull journeys')
+        return
+
+    logger.debug('Trying to culling the journeys')
+
+    """
+    To create a candidates pool, we choose only journeys that are NOT tagged as 'comfort' and 'best' and we create a
+    section set from that pool
+
+    Ex:
+    Journey_1 (Best): Line 14 -> Line 8 -> Bus 172
+    Journey_2 : Line 14 -> Line 6 -> Bus 165
+    Journey_3 : Line 14 -> Line 8 -> Bus 165
+
+    The candidate pool will be like [Journey_2, Journey_3]
+    The sections set will be like set([Line 14, Line 6, Line 8, Bus 165])
+    """
+    candidates_pool, sections_set = _build_candidate_pool_and_sections_set(resp)
+
+    nb_journeys_must_have = len(resp.journeys) - len(candidates_pool)
+    logger.debug("There are {0} journeys we mush keep".format(nb_journeys_must_have))
+
+    nb_journeys_to_find = request["max_nb_journeys"] - nb_journeys_must_have
```

Comment:
let's say Kraken returns 15 journeys, with 1 best, 1 comfort, which means we have 2 journeys must keep and 13 journeys in candidates_pool.  On the other hand, the user wants 5 journeys in max, so have to find only 3 journeys(`nb_journeys_to_find`) among 13 journeys(`len(candidates_pool)`) in candidate_pool...